### PR TITLE
Zoho/l41 t1668 check params usermanagementcontroller

### PR DESCRIPTION
**Pull request by Trischi80**
Dev: check GET and POST params

### CODE REVIEW

File: application/controllers/SurveyAdministrationController.php [hunk 1]
```diff
@@ -2006,6 +1983,7 @@ public function actionOrganize()
 
         $iSurveyID = $this->getSurveyIdFromGetRequest();
```
Comment (thedirtypanda) [inline application/controllers/SurveyAdministrationController.php:1985]: Maybe you could combine those two lines to only one? `$iSurveyID = sanitize_int($this->getSurveyIdFromGetRequest));`
What do you think?

File: application/controllers/UserManagementController.php [hunk 2]
```diff
@@ -378,8 +394,8 @@ public function actionUserPermissions()
             );
         }
 
-        $oRequest = Yii::app()->request;
-        $userId = $oRequest->getParam('userid');
+        $userId = Yii::app()->request->getParam('userid');
+        $userId = sanitize_int($userId);
```
Comment (thedirtypanda) [inline application/controllers/UserManagementController.php:398]: The same for that one. `$userId = sanitize_int(Yii::app()->request->getParam('userid');` Its just an idea but im not sure. Maybe you have another opinion about that.

File: application/controllers/UserManagementController.php [hunk 3]
```diff
@@ -465,9 +481,8 @@ public function actionUserTemplatePermissions(): ?string
             );
         }
         $aTemplateModels = Template::model()->findAll();
-        $oRequest = Yii::app()->request;
-        $userId = $oRequest->getParam('userid');
-        $oUser = User::model()->findByPk($userId);
+        $userId = Yii::app()->request->getParam('userid');
+        $oUser = User::model()->findByPk((int)$userId);
```
Comment (thedirtypanda) [inline application/controllers/UserManagementController.php:485]: See above.

**Review by thedirtypanda — CHANGES_REQUESTED** [review]

**Comment by Shnoulle** [comment]
If someone try to hack my instance sending 123<script>alert("XSS")</script> : i want to have a log at minimum …

See https://github.com/LimeSurvey/LimeSurvey/pull/2232#issuecomment-1031309986 
> Ensure all login, access control, and server-side input validation failures can be logged with sufficient user context to identify suspicious or malicious accounts and held for enough time to allow delayed forensic analysis.

File: application/controllers/SurveyAdministrationController.php (hunk 1 above)
Comment (Trischi80) [reply application/controllers/SurveyAdministrationController.php:1985]: Yes sure :). Thanks Jessi

**Pull request merged**